test/parser.test.js

```diff
@@ -0,0 +1,236 @@
+
+      const fixture = '<!-- Comment -->'
+      const expected = [ '<!-- Comment -->' ]
+
+      expect(parser(options)(fixture)).to.eql(expected)
+    })
+
+    it('Comment (Content)', function () {
+      const options = { location: false }
+
+      const fixture = '<div><!-- Comment --></div>'
+      const expected = [
+        {
+          tag: 'div',
+          content: [ '<!-- Comment -->' ]
+        }
+      ]
+
+      expect(parser(options)(fixture)).to.eql(expected)
+    })
+  })
+
+  describe('XML', function () {
+    it('CDATA', function () {
+      const options = { xmlMode: true }
+
+      const fixture = '<script><![CDATA[console.log(1);]]></script>'
+      const expected = [
+        {
+          tag: 'script',
+          content: [ 'console.log(1);' ]
+        }
+      ]
+
+      expect(parser(options)(fixture)).to.eql(expected)
+    })
+  })
+
+  describe('Directives', function () {
+    it('Directive', function () {
+      const options = { location: false }
+
+      const fixture = '<?php echo "Hello world"; ?>'
+      const expected = [ '<?php echo "Hello world"; ?>' ]
+
+      expect(parser(options)(fixture)).to.eql(expected)
+    })
+
+    it('Directive (Content)', function () {
+      const options = { location: false }
+
+      const fixture = '<!doctype html><html><?php echo "Hello world"; ?></html>'
+      const expected = [
+        '<!doctype html>',
+        {
+          tag: 'html',
+          content: [ '<?php echo "Hello world"; ?>' ]
+        }
+      ]
+
+      expect(parser(options)(fixture)).to.eql(expected)
+    })
+
+    it('Directive (Custom)', function () {
+      const options = {
+        location: false,
+        directives: [
+          { name: '?=', start: '<', end: '>' }
+        ]
+      }
+
+      const fixture = '<?= echo "Hello word"; ?>'
+      const expected = [ '<?= echo "Hello word"; ?>' ]
+
+      expect(parser(options)(fixture)).to.eql(expected)
+    })
+  })
+
+  describe('Tags', function () {
+    it('Doctype', function () {
+      const options = { location: false }
+
+      const fixture = '<!doctype html><html></html>'
+      const expected = [
+        '<!doctype html>',
+        { tag: 'html' }
+      ]
+
+      expect(parser(options)(fixture)).to.eql(expected)
+    })
+
+    it('Attrs', function () {
+      const options = { location: false }
+
+      const fixture = '<div id="id" class="class"></div>'
+      const expected = [
+        {
+          tag: 'div',
+          attrs: { id: 'id', class: 'class' }
+        }
+      ]
+
+      expect(parser(options)(fixture)).to.eql(expected)
+    })
+
+    it('Attrs', function () {
+      const options = { location: false }
+
+      const fixture = '<div id="id" class="class"></div>'
+      const expected = [
+        {
+          tag: 'div',
+          attrs: { id: 'id', class: 'class' }
+        }
+      ]
+
+      expect(parser(options)(fixture)).to.eql(expected)
+    })
+
+    it('selfClosing', function () {
```

Comment:
What is being tested?

Reply:
https://github.com/posthtml/posthtml-parser/pull/21#discussion-diff-157897709R102, but I need to revisit this PR one more time (today) anyways, currently working on finishing/updating https://github.com/posthtml/posthtml-include/pull/16 so this can finally land soon :)